src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1217,94 +1140,107 @@ protected void shutdownService() {
       this.runningThread = null;
       runningThread.interrupt();
     }
-    
-    /**
-     * Called when the queue has been updated and we may need to wake up the consumer thread.
-     */
-    protected void handleQueueUpdate() {
-      if (! startIfNotStarted()) {
-        Thread currRunningThread = runningThread;
-        if (currRunningThread != null) {
-          LockSupport.unpark(currRunningThread);
-        }
-      }
-    }
 
     @Override
     public void run() {
+      Worker worker = null;
       while (runningThread != null) {
         try {
-          TaskWrapper nextTask = getNextTask();
-          if (nextTask != null) {
-            switch (queuePriority) {
-              case High:
-                workerPool.runHighPriorityTask(nextTask);
-                break;
-              case Low:
-                workerPool.runLowPriorityTask(nextTask);
-                break;
-              default:
-                throw new UnsupportedOperationException();
+          worker = workerPool.getWorker();
+          if (worker != null) {
```

Comment:
have you messed around with switching the order here at all?  not sure why it would matter but might be worth a few benchmarks (grabbing the nextTask before the worker.